app/controllers/api/v1/base_controller.rb

```diff
@@ -1,5 +1,6 @@
-
 class Api::V1::BaseController < ApplicationController
+  load_and_authorize_resource
+
   before_action :retrieve_source_model_by_id, only: [:edit, :update, :destroy, :show]
```

Comment:
I don't think you need this now that you are using cancancan. You should be able to refer to the model as `resource`